src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java

```diff
@@ -78,6 +80,45 @@ public UserDTO getUser(@ApiParam(value = "Id of desired user.", required = true)
                 .map(user -> mapper.map(user, UserDTO.class));
     }
 
+    @GetMapping("/config/user-data")
+    @PreAuthorize("hasRole('USER')")
+    @ApiOperation(value = "Retrieve logged in user", response = UserDTO.class, notes = "Required role: USER")
+    @ApiResponses({@ApiResponse(code = 200, message = "User was successfully retrieved."),
+            @ApiResponse(code = 404, message = "Given user not found.", response = ErrorResponse.class)})
+    public UserDTO getUser(Principal principal) {
+        return mapper.map(userService.findUserByEmail(principal.getName()), UserDTO.class);
+    }
+
+    @PostMapping("/config/change-password")
+    @PreAuthorize("hasRole('USER')")
+    @ApiOperation(value = "Change logged in user password", response = UserDTO.class, notes = "Required role: USER")
+    @ApiResponses({@ApiResponse(code = 200, message = "User password was successfully changed."),
+            @ApiResponse(code = 404, message = "Failed to change user password.", response = ErrorResponse.class)})
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "password", dataType = "string", paramType = "body",
+                    value = "New password"),
+    })
+    public UserDTO changePassword(@RequestParam String password, Principal principal) {
+         User user = userService.findUserByEmail(principal.getName());
+         return mapper.map(userService.changeUserPassword(user.getId(), password), UserDTO.class);
+    }
+
+    @PostMapping("/config/user-data")
+    @PreAuthorize("hasRole('USER')")
+    @ApiOperation(value = "Change logged in user first and last name", response = UserDTO.class, notes = "Required role: USER")
+    @ApiResponses({@ApiResponse(code = 200, message = "User first and last name was successfully changed."),
+            @ApiResponse(code = 404, message = "Failed to change user first and last name.", response = ErrorResponse.class)})
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "firstName", dataType = "string", paramType = "body",
+                    value = "First Name"),
+            @ApiImplicitParam(name = "lastName", dataType = "string", paramType = "body",
+                    value = "Last Name"),
+    })
+    public UserDTO changeDetails(@RequestParam String firstName, @RequestParam String lastName, Principal principal) {
```

Comment:
Co jeśli firstName/lastName = null lub " ". Stwórz nowy obiekt DTO lub użyj istniejącego jeśli to możliwe.